src/PostgREST/Main.hs

```diff
@@ -17,12 +18,18 @@ import Network.Wai.Middleware.Static (staticPolicy, only)
 import Network.Wai.Middleware.RequestLogger (logStdout)
 import Data.List (intercalate)
 import Data.Version (versionBranch)
+import Data.Functor.Identity
+import Data.Text(unpack)
 import qualified Hasql as H
 import qualified Hasql.Postgres as P
 import Options.Applicative hiding (columns)
 
 import PostgREST.Config (AppConfig(..), argParser, corsPolicy)
 
+isServerVersionSupported = do
+  Identity row <- H.tx Nothing $ H.singleEx $ [H.stmt|SHOW server_version_num|]
+  return $ read (unpack row) > 90200
```

Comment:
The only problem with this approach here is that the unpack is being used to infer the type of row. So removing it would imply using ScopedTypeVariables (or something to that effect):
```haskell
  Identity (row :: Text) <- H.tx Nothing $ H.singleEx $ [H.stmt|SHOW server_version_num|]
  return $ read (cs row) > 90200
```